### PR TITLE
add code to sig_compare function to find best match

### PR DESCRIPTION
**Pull request by Gtnick1241**
add code to sig_compare function to find best match.

### CODE REVIEW

File: R/compare_results.R [hunk 1]
```diff
@@ -44,7 +44,12 @@ sig_compare <- function(sig1, sig2, metric = c("cosine", "jsd"),
   comparison$y_sig_index <- as.numeric(comparison$y_sig_index)
   comparison <- comparison[order(comparison[[metric_name]], decreasing = TRUE), 
                            ]
```
Comment (joshua-d-campbell) [inline R/compare_results.R:46]: You should try to keep the brackets "]" on the same line. You can automatically format the code in Rstudio.

File: R/compare_results.R [hunk 2]
```diff
@@ -44,7 +44,12 @@ sig_compare <- function(sig1, sig2, metric = c("cosine", "jsd"),
...
-  return(comparison)
+  split_comparison <- split(comparison, comparison$x_sig_name)
```
Comment (joshua-d-campbell) [inline R/compare_results.R:47]: We still want to be able to do the old way of plotting as well. Could you add a parameter to this function like this:

`type = c("best", "threshold")`

If "best" is selected, then you can run your new code. If "threshold" is selected, you can run the old way using the threshold parameter. Note that you will also have to add this parameter to the other compare-* functions

Comment (joshua-d-campbell) [reply R/compare_results.R:47]: Also, make sure to document the new parameter!

**Review by joshua-d-campbell — CHANGES_REQUESTED** [review]
Nice job overall! A few more things need to be tweaked.

**Comment by Gtnick1241** [comment]
I had already committed a new code.

**Comment by Gtnick1241** [comment]
I am sorry to commit many irrelevant data. My major code is in main_class and standard_table file.

**Pull request closed without merging**